tests/generic_config_updater/test_pfcwd_interval.py

```diff
@@ -0,0 +1,135 @@
+    delete_tmpfile(duthost, config_tmpfile)
+    config_reload(duthost)
+
+    logger.info("TEARDOWN COMPLETED")
+
+
+def prepare_pfcwd_interval_config(duthost, value):
+    """
+    Prepares config db by setting pfcwd poll interval to specified value. If value is empty string, delete the current entry. 
+
+    Args:
+        duthost: DUT host object
+        value: poll interval value to be set
+    """
+
+    logger.info("Setting configdb entry pfcwd poll interval to value: {}".format(value))
+   
+    if value:
+        cmd = "sudo pfcwd interval {}".format(value)
+    else:
+        cmd = "sonic-db-cli CONFIG_DB del \PFC_WD\GLOBAL\POLL_INTERVAL"
+
+
+def get_detection_restoration_times(duthost):
+    """
+    Returns detection_time, restoration_time for an interface. Poll_interval must be greater than both in order to be valid
+    
+    Args:
+        duthost: DUT host object
+    """
+     
+    pfcwd_config = duthost.shell("show pfcwd config")
+    pytest_assert(not pfcwd_config['rc'], "Unable to read pfcwd config")
+    
+    for line in pfcwd_config['stdout_lines']:
+        if line.startswith('Ethernet'):
+            interface = line.split()[0] #Since line starts with Ethernet, we can safely use 0 index
+
+            cmd = "sonic-db-cli CONFIG_DB hget \"PFC_WD|{}\" \"detection_time\" ".format(interface)
+            output = duthost.shell(cmd, module_ignore_errors=True)
+            pytest_assert(not output['rc'], "Unable to read detection time")
+            detection_time = output["stdout"]
+            
+            cmd = "sonic-db-cli CONFIG_DB hget \"PFC_WD|{}\" \"restoration_time\" ".format(interface)
+            output = duthost.shell(cmd, module_ignore_errors=True)
+            pytest_assert(not output['rc'], "Unable to read restoration time")
+            restoration_time = output["stdout"]
+
+            return int(detection_time), int(restoration_time)
+
+    pytest_assert(True, "Failed to read detection_time and/or restoration time")
+            
+
+def get_new_interval(duthost, is_valid):
+    """
+    Returns new interval value for pfcwd poll interval, based on the operation being performed
+
+    Args:
+        duthost: DUT host object
+        is_valid: if is_valid is true, return a valid new interval. Config update should succeed. If is_valid is false, return an invalid new interval. Config update should fail.
+    """
+
+    detection_time, restoration_time = get_detection_restoration_times(duthost)
+    if is_valid:
+        return max(detection_time, restoration_time) + 10
+    else:
+        return min(detection_time, restoration_time) - 10
+
+
+@pytest.mark.parametrize("operation", ["add", "replace", "remove"])
+@pytest.mark.parametrize("field_pre_status", ["existing", "nonexistent"])
+@pytest.mark.parametrize("is_valid_config_update", [True, False])
+def test_pfcwd_interval_config_updates(duthost, ensure_dut_readiness, operation, field_pre_status, is_valid_config_update):
```

Comment:
>test_pfcwd_interval_config_updates [](http://example.com/codeflow?start=4&length=34)

Did you test the interval range which is supported by `pdfwd` command?

Reply:
Yes, the function `def get_new_interval(duthost, is_valid)` takes `is_valid` parameter, which determines whether we are testing supported `pfcwd` command interval or not